include/swift/AST/DiagnosticsSema.def

```diff
@@ -2662,6 +2662,8 @@ ERROR(inheritance_from_cf_class,none,
 ERROR(inheritance_from_objc_runtime_visible_class,none,
       "cannot inherit from class %0 because it is only visible via the "
       "Objective-C runtime", (Identifier))
+ERROR(composition_of_protocol_and_AnyObject,none,
+      "Non-protocol %0 cannot conform to composition %1. Composition %1 involves 'AnyObject'", (Identifier, Type))
```

Comment:
While I appreciate your good punctuation and grammar here, diagnostics have a distinct formatting style. They

- don't begin with capital letters
- rarely involve multiple sentences - when they run on, we use semicolons or break the second part into a note.

Reply:
We have [this document](https://github.com/apple/swift/blob/main/docs/Diagnostics.md) to lay out the common conventions.

Reply:
We can do something like `non-protocol type %0 cannot conform to composition type %1` and omit the `Composition %1 involves 'AnyObject'` part.

What do you think?

Reply:
I still think we ought to highlight the `AnyObject` part of the composition in the message in some way. What might help is to think about this in isolation and come up with a "headline". That's usually a good way to write a diagnostic.